src/grammar.pest

```diff
@@ -18,8 +18,9 @@ char = {
 }
 
 union =  { "[" ~ unionElement ~ ("," ~ unionElement)* ~ "]" }
-unionElement = _{ unionChild }
+unionElement = _{ unionChild | unionArrayIndex }
```

Comment:
If you are not planning to implement general slices (`start:end` or `start:end:step`) in this PR, it would be good to add a "TODO".